packages/validator/src/services/attestationDuties.ts

```diff
@@ -63,12 +63,25 @@ export class AttestationDutiesService {
 
     if (metrics) {
       metrics.attesterDutiesCount.addCollect(() => {
+        const currentSlot = this.clock.getCurrentSlot();
         let duties = 0;
-        for (const attDutiesAtEpoch of this.dutiesByIndexByEpoch.values()) {
+        let nextDutySlot = null;
+        for (const [epoch, attDutiesAtEpoch] of this.dutiesByIndexByEpoch.entries()) {
```

Comment:
is `dutiesByIndexByEpoch.keys()` sorted by epoch? normally we store duties of current epoch and next epoch, I think we only need to scan through the 1st epoch to find `nextDutySlot`

Reply:
> is dutiesByIndexByEpoch.keys() sorted by epoch?

Yes that is sorted but there are still few issues if we would just look at the 1st epoch
- it will be a historical epoch (not yet pruned), the next duty slot must be the closest future slot
https://github.com/ChainSafe/lodestar/blob/4580730ca55e4a691e1606955e5048e08637a4c8/packages/validator/src/services/attestationDuties.ts#L14
- we have to only consider future slots, the `duty.slot` needs to be checked against `currentSlot` and since we do not know at which slot in the epoch this is called due to how `addCollect` works, the next duty slot might already be in the next epoch
https://github.com/ChainSafe/lodestar/blob/4580730ca55e4a691e1606955e5048e08637a4c8/packages/validator/src/services/attestationDuties.ts#L77
- there is a edge case when a validator becomes active, it can happen that `dutiesByIndex` is empty